### PR TITLE
feat: update .NET SDK to use Teams JS V2

### PR DESCRIPTION
**Pull request by yiqing-zhao**
E2E TEST: N/A

### CODE REVIEW

**Review by JerryYangKai — DISMISSED** [review]

**Review by yukun-dong — APPROVED** [review]

File: packages/dotnet-sdk/README.md [hunk 1]
```diff
@@ -391,20 +418,40 @@ private void printLog(LogLevel level, string message)
...
+
 1. Right click project file and choose "Manage User Secrets".
 2. Change key name "CLIENT_SECRET" to "TeamsFx:Authentication:ClientSecret".
 
+### Upgrade from 1.1.0 to 1.2.0 (Update projects to use TeamsJS V.2.0)
+
+Teams Toolkit provides users with a template `TeamsJSBlazorInterop.js`, which consists of multiple commonly used Teams JS SDK API. Users can add more APIs if needed. As suggested, even if you intend your app to only run in Teams (and not Office and Outlook), best practice is to start referencing the latest TeamsJS (_v.2.0_ or later) as soon as convenient, in order to benefit from the latest improvements, new features, and support (even for Teams-only apps).
+
+Starting from TeamsFx .NET SDK 1.3.0, TeamsJS V2 are referenced. Though previously scaffolded projects still work with TeamsFx .NET SDK 1.3.0, we suggest you replace the `TeamsJSBlazorInterop.js` under `./wwwroot/js` with the latest one from [here](https://github.com/OfficeDev/TeamsFx/blob/main/templates/tab/csharp/default/wwwroot/js/TeamsJsBlazorInterop.js). Two APIs in the file are updated with new function name to align with renamed TeamsJS APIs. `initializeWithContext()` is removed.
```
Comment (tecton) [inline packages/dotnet-sdk/README.md:440]: If we are going to release in 1.2.0, this need to be changed.

File: packages/dotnet-sdk/src/TeamsFx/Credential/TeamsUserCredential.cs [hunk 2]
```diff
@@ -235,7 +235,7 @@ private async ValueTask<IJSObjectReference> ImportTeamsSdk(IJSRuntime jsRuntime)
     {
         try
         {
-            await jsRuntime.InvokeVoidAsync("import", "https://statics.teams.cdn.office.net/sdk/v1.10.0/js/MicrosoftTeams.min.js").ConfigureAwait(false);
+            await jsRuntime.InvokeVoidAsync("import", "https://res.cdn.office.net/teams-js/2.0.0/js/MicrosoftTeams.min.js").ConfigureAwait(false);
```
Comment (tecton) [inline packages/dotnet-sdk/src/TeamsFx/Credential/TeamsUserCredential.cs:238]: Can we use the latest version of teams-js? It's 2.4.2 now.

File: packages/dotnet-sdk/README.md (hunk 1 above)
Comment (yiqing-zhao) [reply packages/dotnet-sdk/README.md]: Thanks! I've changed 1.3.0 -> 1.2.0.

File: packages/dotnet-sdk/src/TeamsFx/Credential/TeamsUserCredential.cs (hunk 2 above)
Comment (yiqing-zhao) [reply packages/dotnet-sdk/src/TeamsFx/Credential/TeamsUserCredential.cs]: Updated. Thanks!

**Review by tecton — APPROVED** [review]

**Review by hund030 — APPROVED** [review]

**Review by Siglud — APPROVED** [review]

**Pull request merged**